src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -841,38 +842,49 @@ template <size_t SpatialDim, typename Frame>
 struct ConstraintGamma0Compute : ConstraintGamma0, db::ComputeTag {
   using argument_tags =
       tmpl::list<domain::Tags::Coordinates<SpatialDim, Frame>>;
-  static auto function(
+
+  using return_type = Scalar<DataVector>;
+
+  static constexpr void function(
+      const gsl::not_null<Scalar<DataVector>*> gamma,
       const tnsr::I<DataVector, SpatialDim, Frame>& coords) noexcept {
-    const DataVector r_squared = get(dot_product(coords, coords));
-    Scalar<DataVector> gamma = make_with_value<type>(coords, 0.);
-    get(gamma) = 3. * exp(-0.0078125 * r_squared) + 0.001;
-    return gamma;
+    get(*gamma) =
+        3. * exp(-0.0078125 * get(dot_product(coords, coords))) + 0.001;
   }
+
   using base = ConstraintGamma0;
 };
 /// \copydoc ConstraintGamma0Compute
 template <size_t SpatialDim, typename Frame>
 struct ConstraintGamma1Compute : ConstraintGamma1, db::ComputeTag {
   using argument_tags =
       tmpl::list<domain::Tags::Coordinates<SpatialDim, Frame>>;
-  static constexpr auto function(
+
+  using return_type = Scalar<DataVector>;
+
+  static constexpr void function(
+      const gsl::not_null<Scalar<DataVector>*> gamma1,
       const tnsr::I<DataVector, SpatialDim, Frame>& coords) noexcept {
-    return make_with_value<type>(coords, -1.);
+    destructive_resize_components(gamma1, get<0>(coords).size());
```

Comment:
You inconsistently do the destructive_resize in the hpps. Could you use it in the other compute tags that have the function defined in line?

Reply:
I added it to all functions. I thought the assignment in the function implementation would make the resize automatically? If not, I'm afraid there are several other functions in GR/GH that might require the manual `destructive_resize` :P

Reply:
Yes, the assignment does the check. Either is fine, though it would be nice for them to (eventually) be consistent :)